### PR TITLE
jaxb-tck/issues/78 switch to https://www.w3.org/2005/05/xmlmime

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>

This change doesn't seem to help directly with https://github.com/eclipse-ee4j/jaxb-tck/issues/78 but will create a draft pr in case we want to merge it anyway (we can switch to ready for review).  I immediately saw the following failures with this change, so it didn't seem to help:
```
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcInherit5.html#testCase0001 Failed. test cases: 1; all failed; first test case failure: compileSchema
Pass: 181  Fail: 1  Error: 0  Not-Run: 24,444    
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcInherit5.html#testCase0002 Failed. test cases: 1; all failed; first test case failure: compileSchema
Pass: 181  Fail: 2  Error: 0  Not-Run: 24,443    
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcInherit5.html#testCase0003 Failed. test cases: 1; all failed; first test case failure: compileSchema
Pass: 181  Fail: 3  Error: 0  Not-Run: 24,442    
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcNone.html#testCase0001 Failed. test cases: 1; all failed; first test case failure: compileSchema
Pass: 181  Fail: 4  Error: 0  Not-Run: 24,441    
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcNone.html#testCase0002 Failed. test cases: 1; all failed; first test case failure: compileSchema
Pass: 181  Fail: 5  Error: 0  Not-Run: 24,440    
Sep 13, 2022, 3:09:54 PM Finished api/jakarta_xml/bind/JAXBContext/java2schema/CustomizedMapping/packages/XmlAccessorType/srcNone.html#testCase0003 Failed. test cases: 1; all failed; first test case failure: compileSchema
```

### CODE REVIEW

File: jaxb-ri/core/src/main/java/org/glassfish/jaxb/core/v2/WellKnownNamespace.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ private WellKnownNamespace() {} // no instanciation please
 
     public static final String SWA_URI = "http://ws-i.org/profiles/basic/1.1/xsd";
 
-    public static final String XML_MIME_URI = "http://www.w3.org/2005/05/xmlmime";
+    public static final String XML_MIME_URI = "https://www.w3.org/2005/05/xmlmime";
```
Comment (bstansberry) [inline jaxb-ri/core/src/main/java/org/glassfish/jaxb/core/v2/WellKnownNamespace.java:26]: I believe this shouldn't change. It's the namespace name, which is just a string. And that name is part of the spec.

File: jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/schemagen/XmlSchemaGenerator.java [hunk 2]
```diff
@@ -622,7 +622,7 @@ private void writeTo(Result result, Map<Namespace,String> systemIds) throws IOEx
                     schema._import().namespace(WellKnownNamespace.SWA_URI).schemaLocation("http://ws-i.org/profiles/basic/1.1/swaref.xsd");
                 }
                 if(useMimeNs) {
-                    schema._import().namespace(WellKnownNamespace.XML_MIME_URI).schemaLocation("http://www.w3.org/2005/05/xmlmime");
+                    schema._import().namespace(WellKnownNamespace.XML_MIME_URI).schemaLocation("https://www.w3.org/2005/05/xmlmime");
```
Comment (bstansberry) [inline jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/schemagen/XmlSchemaGenerator.java:625]: This schema location otoh is a url where the schema is found, so there changing https makes sense to me.

Comment (scottmarlow) [reply jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/schemagen/XmlSchemaGenerator.java:625]: Thanks, I'll try to test again without this line changed!

**Comment by scottmarlow** [comment]
Results so far are looking better:
```
Pass: 787  Fail: 0  Error: 0  Not-Run: 23,839    
Sep 13, 2022, 6:01:22 PM Finished schema_bind/bind_globalBindings/elem_serializable/elem_serializable00101m1.html#elem_serializable00101m1_p Passed. test cases: 3; all passed

Pass: 788  Fail: 0  Error: 0  Not-Run: 23,838    
Sep 13, 2022, 6:01:23 PM Finished schema_bind/bind_globalBindings/elem_serializable/elem_serializable00102m1.html#elem_serializable00102m1 Passed.
Pass: 789  Fail: 0  Error: 0  Not-Run: 23,837    
Sep 13, 2022, 6:01:23 PM Finished schema_bind/bind_globalBindings/elem_serializable/elem_serializable00102m1.xsd.html#elem_serializable00101m1.xml Passed. test cases: 1; all passed
Pass: 790  Fail: 0  Error: 0  Not-Run: 23,836    
Sep 13, 2022, 6:01:23 PM Finished schema_bind/bind_globalBindings/elem_serializable/elem_serializable00102m1.html#elem_serializable00102m1_p Passed. test cases: 3; all passed
Pass: 791  Fail: 0  Error: 0  Not-Run: 23,835    
Sep 13, 2022, 6:01:24 PM Finished schema_bind/bind_globalBindings/enableJavaNamingConventions/AD_valConstr00101m_Bind2.bind.xsd.html#AD_valConstr00101m1_p.xml Passed. test cases: 1; all passed
Pass: 792  Fail: 0  Error: 0  Not-Run: 23,834    
Sep 13, 2022, 6:01:24 PM Finished schema_bind/bind_globalBindings/enableJavaNamingConventions/AD_valConstr00101m_Bind2.html#AD_valConstr00101m1_Bind2_n Passed. test cases: 3; all passed
Pass: 793  Fail: 0  Error: 0  Not-Run: 23,833    
Sep 13, 2022, 6:01:24 PM Finished schema_bind/bind_globalBindings/enableJavaNamingConventions/AD_valConstr00101m_Bind2.html#AD_valConstr00101m_Bind2 Passed.
Pass: 794  Fail: 0  Error: 0  Not-Run: 23,832    
Sep 13, 2022, 6:01:24 PM Finished schema_bind/bind_globalBindings/enableJavaNamingConventions/AD_valConstr00101m_Bind2.html#AD_valConstr00101m1_Bind2_p Passed. test cases: 3; all passed

Pass: 795  Fail: 0  Error: 0  Not-Run: 23,831    
```

Will let it run to completion and report back on results.

**Comment by scottmarlow** [comment]
Looks good now, XML Binding 4.0 TCK Test results from WildFly with this change:
```
ass: 24,626  Fail: 0  Error: 0  Not-Run: 0    
Test results: passed: 24,626
```

**Comment by jimma** [comment]
> Looks good now, XML Binding 4.0 TCK Test results from WildFly with this change:
> 
> ```
> ass: 24,626  Fail: 0  Error: 0  Not-Run: 0    
> Test results: passed: 24,626
> ```

@scottmarlow  It's good to know that these uris are actually defined in jaxb-ri instead of jaxb-tck source.

**Review by lukasj — APPROVED** [review]
LGTM

**Comment by lukasj** [comment]
4.0.1 is in staging

**Pull request merged**